zebrad/src/components/sync.rs

```diff
@@ -172,61 +166,30 @@ where
         self.request_genesis().await?;
 
         'sync: loop {
-            // Update metrics for any ready tasks, before wiping state
-            while let Some(Some(rsp)) = self.pending_blocks.next().now_or_never() {
-                match rsp.expect("block download and verify tasks should not panic") {
-                    Ok(hash) => tracing::trace!(?hash, "verified and committed block to state"),
-                    Err((e, _)) => {
-                        tracing::trace!(?e, "sync error before restarting sync, ignoring")
-                    }
-                }
-            }
-            self.update_metrics();
-
             // Wipe state from prevous iterations.
             self.prospective_tips = HashSet::new();
-            self.pending_blocks = Box::pin(FuturesUnordered::new());
+            self.downloads.cancel_all();
```

Comment:
![cancelled](https://media1.popsugar-assets.com/files/thumbor/E14x2plpHZy-Bk_1kC4jaJ0J7aA/fit-in/2048xorig/filters:format_auto-!!-:strip_icc-!!-/2016/12/08/871/n/1922507/a8febd43d66232e0_cancelled/i/Canceled.gif)